### DISCUSSION TITLE
Environment Variable values

### DISCUSSION THREAD

**Discussion by mitski-shinm** [Qmirdev/EthelMC#3] [Q&A] [answered]
Is it possible to update the method of setting values in the app.py file by retrieving them from environment variables instead?
for example Bot Token and owner ids?

**Comment by q31r** [comment, marked as answer]
Yes, for some sensitive informations we can get the variable values from environment variables in `app.py`, you can use the os module in Python:
```python
import os

BOT_TOKEN = os.environ['BOT_TOKEN'] 
OWNER_UID = [os.environ['OWNER_UID']]  

SERVER_NAME = os.environ['SERVER_NAME']
SERVER_ADDRESS = os.environ['SERVER_ADDRESS']
SERVER_PORT = os.environ['SERVER_PORT']
SERVER_STORE = os.environ['SERVER_STORE']

# Rest of the code
```

So instead of hardcoding the values, you load them from environment variables at runtime. This allows you to configure the values externally without changing code ☺️
